samples/cellular/nrf_cloud_mqtt_multi_service/src/cloud_connection.c

```diff
@@ -182,40 +205,64 @@ static void cloud_ready(void)
  */
 static void update_shadow(void)
 {
+	static bool updated;
 	int err;
 	struct nrf_cloud_svc_info_fota fota_info = {
 		.application = nrf_cloud_fota_is_type_enabled(NRF_CLOUD_FOTA_APPLICATION),
 		.bootloader = nrf_cloud_fota_is_type_enabled(NRF_CLOUD_FOTA_BOOTLOADER),
 		.modem = nrf_cloud_fota_is_type_enabled(NRF_CLOUD_FOTA_MODEM_DELTA),
 		.modem_full = nrf_cloud_fota_is_type_enabled(NRF_CLOUD_FOTA_MODEM_FULL)
 	};
-
 	struct nrf_cloud_svc_info_ui ui_info = {
 		.gnss = location_tracking_enabled(),
 		.temperature = IS_ENABLED(CONFIG_TEMP_TRACKING),
-		.log = IS_ENABLED(CONFIG_NRF_CLOUD_LOG_BACKEND) &&
-		       IS_ENABLED(CONFIG_LOG_BACKEND_NRF_CLOUD_OUTPUT_TEXT),
+		.log = (IS_ENABLED(CONFIG_NRF_CLOUD_LOG_BACKEND) &&
+		        IS_ENABLED(CONFIG_LOG_BACKEND_NRF_CLOUD_OUTPUT_TEXT)) ||
+		       (IS_ENABLED(CONFIG_NRF_CLOUD_LOG_DIRECT) &&
+			!IS_ENABLED(CONFIG_LOG_BACKEND_NRF_CLOUD_OUTPUT_DICTIONARY)),
 		.dictionary_log = IS_ENABLED(CONFIG_NRF_CLOUD_LOG_BACKEND) &&
 				  IS_ENABLED(CONFIG_LOG_BACKEND_NRF_CLOUD_OUTPUT_DICTIONARY)
 	};
 	struct nrf_cloud_svc_info service_info = {
 		.fota = &fota_info,
 		.ui = &ui_info
 	};
+#if defined(CONFIG_NRF_MODEM)
+	struct nrf_cloud_modem_info modem_info = {
+		.device = NRF_CLOUD_INFO_SET,
+		.network = NRF_CLOUD_INFO_SET,
+		.sim = IS_ENABLED(CONFIG_MODEM_INFO_ADD_SIM) ? NRF_CLOUD_INFO_SET : 0,
+		.mpi = NULL,
+		/* Include the application version */
+		.application_version = CONFIG_APP_VERSION
+	};
+	struct nrf_cloud_modem_info *mdm = &modem_info;
+#else
+	struct nrf_cloud_modem_info *mdm = NULL;
+#endif
 	struct nrf_cloud_device_status device_status = {
-		.modem = NULL,
+		.modem = mdm,
```

Comment:
I'm not sure what you mean by "nrf_cloud can query this information itself."  From where?

I didn't design the device_info struct.  We can discuss redesigning it in the future.

This isn't "relying" on CONFIG_NRF_MODEM.  It will build with it disabled.

Reply:
However, it is a better user experience in the nRF Cloud portal if these areas are updated to reflect the name of the sample and how it is connected, rather than leave stale information there (potentially).

Reply:
we have [CONFIG_NRF_CLOUD_SEND_DEVICE_STATUS](https://github.com/nrfconnect/sdk-nrf/blob/b712ba60fb59d184523f3da0a8265ceea758cf6b/subsys/net/lib/nrf_cloud/Kconfig#L46) and its related config options.  so this information is already sent on connect for M(QTT)MSS.
can they be implemented for CoAP too, so the application doesn't have to do this?
if not, they can be moved into the mqtt specific kconfig.

Reply:
OK, I can rework the CoAP library to do this, but I'd like to do it in a follow-on PR.

Reply:
https://nordicsemi.atlassian.net/browse/IRIS-6792